src/Widgets/DXPCompanyDetails/DXPCompanyDetailsWidgetComponent.tsx

```diff
@@ -14,14 +14,31 @@ provideComponent(DXPCompanyWidget, () => {
   const { register, handleSubmit } = useForm()
 
   const onDeleteClick = () => {
-    company?.delete()
-    navigateTo(Obj.getByPermalink('organizations-page'))
+    return company?.delete().catch((error: ClientError) => {
+      onError(error)
+    }).finally(() => {
+      navigateTo(Obj.getByPermalink('organizations-page'))
+    })
```

Comment:
We shouldn't navigate back to the previous page if the delete was not successful, so the user can try again